crates/vm2-interface/src/state_interface.rs

```diff
@@ -1,93 +1,148 @@
 use primitive_types::{H160, U256};
 
+/// Public interface of the VM state. Encompasses both read and write methods.
 pub trait StateInterface {
+    /// Reads a register with the specified zero-based index. Returns a value together with a pointer flag.
     fn read_register(&self, register: u8) -> (U256, bool);
+    /// Sets a register with the specified zero-based index
     fn set_register(&mut self, register: u8, value: U256, is_pointer: bool);
 
+    /// Returns a mutable handle to the current call frame.
     fn current_frame(&mut self) -> impl CallframeInterface + '_;
+    /// Returns the total number of call frames.
     fn number_of_callframes(&self) -> usize;
+    /// Returns a mutable handle to a call frame with the specified index, where
     /// zero is the current frame, one is the frame before that etc.
     fn callframe(&mut self, n: usize) -> impl CallframeInterface + '_;
 
-    fn read_heap_byte(&self, heap: HeapId, index: u32) -> u8;
-    /// Reads an entire `U256` word in the big-endian order from the specified heap page / `index`
+    /// Reads a single byte from the specified heap page at the specified 0-based offset.
+    fn read_heap_byte(&self, heap: HeapId, offset: u32) -> u8;
+    /// Reads an entire `U256` word in the big-endian order from the specified heap page / `offset`
     /// (which is the index of the most significant byte of the read value).
-    fn read_heap_u256(&self, heap: HeapId, index: u32) -> U256;
-    /// Writes an entire `U256` word in the big-endian order to the specified heap page at the specified `index`
+    fn read_heap_u256(&self, heap: HeapId, offset: u32) -> U256;
+    /// Writes an entire `U256` word in the big-endian order to the specified heap page at the specified `offset`
     /// (which is the index of the most significant byte of the written value).
-    fn write_heap_u256(&mut self, heap: HeapId, index: u32, value: U256);
+    fn write_heap_u256(&mut self, heap: HeapId, offset: u32, value: U256);
 
+    /// Returns current execution flags.
     fn flags(&self) -> Flags;
+    /// Sets current execution flags.
     fn set_flags(&mut self, flags: Flags);
 
+    /// Returns the currently set 0-based transaction number.
     fn transaction_number(&self) -> u16;
+    /// Sets the current transaction number.
     fn set_transaction_number(&mut self, value: u16);
 
+    /// Returns the value of the context register.
     fn context_u128_register(&self) -> u128;
+    /// Sets the value of the context register.
```

Comment:
This comment adds no information, which is why I don't like to have mandatory comments. I would explain the difference between the context register and the context field of the callframe here but I don't remember what the difference is.

Reply:
Mandatory comments looks prettier in the generated docs. Otherwise, I agree that having meaningful method names is sometimes enough 😄